### ISSUE TITLE
[Bug] Misleading UX: Please do not ask users to bug-report slow databases

### ISSUE THREAD

**Issue by mk-pmb** [iv-org/invidious#4159]
**Describe the bug**
Invidious (at least the instance I was using) sometimes displays a message requesting that I bug report their database being slow. This potentially wastes a lot of time and effort for people who are trying to help, just to then find in the FAQ that reporting it here is basically useless. Or even worse, they post it without reading (or understanding) the FAQ and thus waste the time of another volonteer to close the issue.

Let's improve error reporting to check for `DB::PoolTimeout` and in that case, tell users upfront that there's not much to be done, except try another instance or check the instance FAQ for what do do in this case and how to help. We should make it easy for instance operators to customize this message so they can mention ways of supporting them.

**Steps to Reproduce**
Doesn't occurr reliably but a lot: We have tons of `DB::PoolTimeout` issues here in the tracker.

**Logs**

**Screenshots**

**Additional context**

**Comment by unixfox** [comment]
I have always said to @SamantazFox that we should ignore these errors but they said we should keep them so idk...

**Comment by mk-pmb** [comment]
I agree that we should not ignore those errors. That's why I suggested a message that to me seems more productive.

**Comment by unixfox** [comment]
> We can still include the backtrace in a `<details><summary>Technical details for debugging (if you are the instance owner)</summary>…`

People don't care, they will still send a bug report anyway.

**Comment by mk-pmb** [comment]
> People don't care, they will still send a bug report anyway.

Maybe. I wonder if you can measure and compare how many people have instead gone the route described, discovered the FAQ and then just frowned at the fact that invidious has made them waste their time.
Fixing the message may not be 100% effective at reducing spam, but it improves our moral impact and makes us look more respecting of users.

**Comment by SamantazFox** [comment]
> I have always said to @SamantazFox that we should ignore these errors but they said we should keep them so idk...

We _do_ ignore most of the DB issues. The problem is that we use the `DB::ConnectionPool` module to connect to Youtube servers, which causes various `DB::XXXError` to be raised when there is a problem there.

I need to make a proper wrapper for those errors one day....

**Comment by mk-pmb** [comment]
> and is much likely outdated.

I'd be pleasently surprised if it is. I can't currently check whether it's still a problem, but if someone had fixed this, they would probably have said so here.

**Comment by unixfox** [comment]
This has been less and less of a problem since Invidious uses Invidious companion to connect to YouTube servers for the requests that are the most compute intensive.
